### ISSUE TITLE
Wrong error message (typeof with Array.isArray)

### ISSUE THREAD

**Issue by MattMoony** [microsoft/TypeScript#36839]
Even though I am well aware that this bug is not one of the most pressing issues at hand and that it is actually **not a real problem** at all, since all arrays will already produce a truthy value in the first part of the if-clause, I still think that it could lead to some **confusion** at first glance. I tried my best finding a post mentioning this very specific bug, but I couldn't.

**TypeScript Version:**  3.7.5

**Search Terms:** Array.isArray undefined

**Code**

```ts
function foo(bar: number|number[]) {
  if (typeof bar !== 'undefined' || (Array.isArray(bar) && bar.length > 0)) {
    return console.log(':)');
  }
  console.log(':(');
}

foo([1, 2, 3]);
```

**Expected behavior:**
`Error: Property does not exist on type 'never'`

Since `Array.isArray(...)` will, to the best of my knowledge, never produce a truthy value when given something of type `undefined`. However, it should produce an `Error: Property does not exist on type 'never'` error, since the second part of the clause will never be evaluated with something of type `array` (because that would already be truthy in the first half).

**Actual behavior:**
`Error: Object is possibly 'undefined'`

![ts-bug](https://user-images.githubusercontent.com/41878842/74683188-67806c00-51c8-11ea-873d-0d7d0fbda014.PNG)

**Playground Link:** 

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
Duplicate #26914
